distributed/scheduler.py

```diff
@@ -1598,30 +1597,21 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
     def publish_dataset(self, stream=None, keys=None, data=None, name=None,
                         client=None):
+        if name in self.datasets:
+            return {'success': 'false'}
         self.client_wants_keys(keys, 'published-%s' % name)
-        self.published_data[name] = {'data': data, 'keys': keys,
-                                     'clients': {client}}
+        self.datasets[name] = {'data': data, 'keys': keys}
+        return {'success':  'true'}
```

Comment:
Suggest `{'status': 'OK'}`

This is the common response for other routes within the system.